examples/torch/substra-network.py

```diff
@@ -0,0 +1,185 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+import dataclasses
+import os
+import shutil
+import tarfile
+import tempfile
+import uuid
+import zipfile
+
+import substra
+import torchvision
+
+DATASET_ROOT = './data'
+CURRENT_DIR = os.path.dirname(__file__)
+PUBLIC_PERMISSIONS = {'public': True, 'authorized_ids': []}
+
+_CIFAR10_ARCHIVE_NAME = 'cifar-10-python.tar.gz'
+
+# TODO format data to separate X and y?
+# TODO multiple nodes example
+# TODO create a click wrapper around the different commands
+# TODO make this example reliable to run multiple times (could use tag to store ts)
+
+
+@dataclasses.dataclass
+class Context:
+    profile: str = 'node1'
+    assets_root: str = os.path.join(CURRENT_DIR, 'assets')
+
+    _client: substra.Client = None
+
+    @property
+    def client(self):
+        return self._client
+
+    def initialize(self):
+        self._client = substra.Client(profile_name=self.profile)
+        self._client.login()
+        return self
```

Comment:
I feel this is added complexity for code that is no different from have 2 global variables:
```
client = substra.Client(profile_name='node-1')
client.login()
assets_root = os.path.join(CURRENT_DIR, 'assets')
```
If we don't want to have these as global variables, they could be created in the __main__ block and passed as args to register(), no need for an object.